src/js/components/Meter/Meter.js

```diff
@@ -47,12 +52,28 @@ const Meter = forwardRef(
       !(theme.dir === 'rtl' && reverseProp);
 
     const memoizedMax = useMemo(() => deriveMax(values), [values]);
+    const messageId = values && values.length === 1 ? 'singular' : 'plural';
+
+    const meterAriaLabel =
+      ariaLabel ||
+      format({
+        id: `meter.${messageId}`,
+        messages: messages?.meter,
+        values: {
+          meterValue:
+            value || values.map((item) => item.value ?? 0).join(', ') || 0,
+          type,
+          max: max || memoizedMax || 100,
+        },
+      });
+
     let content;
     if (type === 'bar') {
       content = (
         <Bar
           ref={ref}
-          max={memoizedMax}
+          aria-label={meterAriaLabel}
+          max={max || memoizedMax}
```

Comment:
This isn't quite the same but probably an improvement.
Previously if the max prop was in `rest` I think it would override this property. Now you will get `memoizedMax` if max is specified but falsey (e.g. 0, undefined, null). Maybe that's better? That does assume we can't have negative values (e.g. say a range of values from -10 to 0)?

I think it's little fragile that the rules for what max is are now split into 4 places:

1. deriveMax
2. the message max param
3. Bar max property (does logic)
4. Cirlce max property (does logic)

This is alot of risk for disagreement. Right now if there are multiple `values` but they are 0 you will get a `memoizedMax` of 0. The message will say 100 but the Bar and Circle will be passed `max=0`

`memoizedMax` will be 100 or if there are at least 2 `values` it adds them up. That also means if `values` only contains 1 thing it will return 100 or if there's a `value` it will return 100.

To avoid this we might want to refactor to normalize max in one spot and pass that one value to the message, Bar and Circle. Maybe something like:

Change the component prop to be `max: maxProp` and then
```javascript
const max = useMemo(() => {
  let maxValue = 100;
  if (values?.length > 1) {
    maxValue = values.reduce((total, currentValue) => total + currentValue, 0);
  }
  return maxValue || maxProp || 100;
}, [maxProp, values]);  
```
 And then just use `max` for the 3 places it's needed (message param, Bar prop and Circle prop).

Reply:
@MikeKingdom 

yes agree there is alot being done here 

```
deriveMax
the message max param
Bar max property (does logic)
Cirlce max property (does logic)
```

I added the suggestion you provided with a small change. 

1. I needed 
` maxValue = values.reduce((total, currentValue) => total + currentValue, 0);`
currentValue.value

2. I put the `maxProp` to be the first in our || statement so its the following 
`  return maxProp || maxValue || 100;`

for example if I have 
```
      <Meter type="bar" max={40} value={30} />
```
with having maxValue || maxProp || 100 
it was taking the maxValue of 100 and not respecting what the user passed in for `maxProp`